packages/components/src/components/hds/disclosure-primitive/index.ts

```diff
@@ -33,11 +35,42 @@ export interface HdsDisclosurePrimitiveSignature {
 }
 
 export default class HdsDisclosurePrimitiveComponent extends Component<HdsDisclosurePrimitiveSignature> {
-  @tracked isOpen = this.args.isOpen ?? false;
+  @tracked _isOpen = false;
+  @tracked _isControlled = this.args.isOpen !== undefined;
```

Comment:
I asked copilot to provide me an example of a component in Ember that was both controlled and uncontrolled and this is what it came up with:
<img width="811" alt="screenshot_3915" src="https://github.com/hashicorp/design-system/assets/686239/c1ab36fa-7383-4e82-b53f-6de22131389a">
this made me question if we should determine if it's controlled not by the argument `isOpen` but by the fact that consumers would provide an `@onClickToggle` argument to handle/control the state change.
which btw is not very dissimilar from what we're doing here https://github.com/hashicorp/design-system/blob/main/packages/components/src/components/hds/pagination/compact/index.js#L43-L52 

just an extra data input, in case can help in some way

Reply:
I’d say the implementation you’ve landed on is somewhere between controlled and uncontrolled. I’d call it a “latch” — if the caller provides an initial `isOpen` state but not the `onChange` callback, then the component manages its state internally until the caller sends in a new `isOpen` state. Appropriate for this use case, but a little more complex then straight “controlled” (where the caller must also provide `onChange` if they want to control the state).